src/event.rs

```diff
@@ -372,6 +372,14 @@ pub enum WindowEvent<'a> {
     ///
     /// At the moment this is only supported on Windows.
     ThemeChanged(Theme),
+
+    /// The window has been occluded (completely hidden from view).
+    ///
+    /// This is different to window visibility as it depends on whether the window is closed,
+    /// minimised, set invisible, or fully occluded by another window.
+    ///
+    /// At the moment this is only supported on macOS.
```

Comment:
```suggestion
    /// ## Platform-specific
    ///
    /// At the moment this is only supported on macOS.
```

We probably want to put platform-specific notes like this under a `## Platform-specific` heading.